pkg/controllers/provisioning/provisioner.go

```diff
@@ -40,7 +40,7 @@ type Provisioner struct {
 	// State
 	*v1alpha5.Provisioner
 	pods    chan *v1.Pod
-	results chan error
+	results chan struct{}
```

Comment:
After reading up on this topic, I think we might want to use a different strategy here. AFAICT `results` (the way it's being done here) is not a common idiom. Also, although I don't know of any problem specifically, it does seem a bit fragile to rely on "counting" (i.e., the number of writes to the `results` channel needs to match the number of goroutines blocked waiting to read something). I think rather than re-use the `results` channel in the `Provisioner` struct, callers of `Provisioner.Add` should pass in a channel that they will wait on (or pass in a callback that will do something arbitrary, like close a channel, which the caller will just happen to block on). Thoughts?